### PR TITLE
feat: add `nsfw` field to application commands

### LINKED ISSUE #591: Nsfw support for application commands
### Summary

This is a feature request to add the nsfw support for application commands into disnake

### What is the feature request for?

The core library

### The Problem

Implementing the new feature 

### The Ideal Solution

Implementing the new feature (it's just a boolean field in the command object, that feature was mentioned in discord api roadmap) 

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

See today's [announcement](https://canary.discord.com/channels/613425648685547541/697138785317814292/1040031507026284645) in ddevs + https://github.com/discord/discord-api-docs/pull/5617.
Will be released on the 16th, but the api should already (or still) work. Mobile platform support (stable) is planned for the 28th.

Closes #591.
Reverts acda74cc87dbd6e2f43cec9bd0741cdafa076a5a.

#### TODO
- [x] Make sure the api docs links are valid, once they're available.
- [x] Do some more testing.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by lol219** [comment]
The docs are merged ,right?

**Comment by ooliver1** [comment]
> The docs are merged ,right?

Did you try reading the PR description?

File: disnake/app_commands.py [hunk 1]
```diff
@@ -453,13 +453,20 @@ class ApplicationCommand(ABC):
         Defaults to ``True``.
 
         .. versionadded:: 2.5
+
+    nsfw: :class:`bool`
+        Whether this command is :ddocs:`age-restricted <interactions/application-commands#agerestricted-commands>`.
+        Defaults to ``False``.
```
Comment (onerandomusername) [inline disnake/app_commands.py:459]: note that bots in discovery **cannot** have nsfw (app) commands.

Comment (shiftinv) [reply disnake/app_commands.py:459]: The linked api docs will mention this:
https://github.com/discord/discord-api-docs/blob/766f6c5adf3aa14cc0ee6a7dca5ef3f9532dc534/docs/interactions/Application_Commands.md?plain=1#L1024

**Review by onerandomusername — APPROVED** [review]
tested previously, forgot to approve

**Pull request merged**